gateway-ha/src/main/java/io/trino/gateway/ha/router/QueryCountBasedRouter.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.gateway.ha.router;
+
+import io.trino.gateway.ha.clustermonitor.ClusterStats;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class QueryCountBasedRouter
+        extends StochasticRoutingManager
+{
+    private static final Logger log = LoggerFactory.getLogger(QueryCountBasedRouter.class);
+    private List<LocalStats> clusterStats;
+
+    List<LocalStats> clusterStats()
+    {
+        return clusterStats;
+    }
```

Comment:
Let's prevent someone misuse this function in the future.
```suggestion
    @VisibleForTesting
    synchronized List<LocalStats> clusterStats()
    {
        return new ArrayList<>(clusterStats);
    }
```